### ISSUE TITLE
`$X` should be enforced to be equal across multiple patterns in the same rule

### ISSUE THREAD

**Issue by ievans** [returntocorp/semgrep#176]
Currently, if we have a rule with multiple patterns, the metavariable (say $X) does not need to be the same across the patterns in that rule. So:

```
patterns:
    pattern-inside: def foo($X): ...
    pattern: $X = 1
```

will match:
```
def foo(x):
    y = 1
```
but it would be more intuitive to only match:
```
def foo(x):
    x = 1
```

It may be possible to do this only in sgrep.py

**Comment by DrewDennison** [comment]
Two case on sgrep.live

https://sgrep.live/o5d

and https://sgrep.live/6xY (which currently exposes a bug)

**Comment by nbrahms** [comment]
Blocked on https://github.com/returntocorp/sgrep/issues/328

**Comment by aryx** [comment]
This should be unblocked now.

**Comment by nbrahms** [comment]
Fixed by #712 :)
